Allows creation of working clusters inside testermint

patimen opened the pull request:
Uses the DockerClient to create the network of nodes for testing. Also, get rid of the http server, as Wiremock can serve the exact same purpose.

patimen committed:
Allows creation of working clusters inside testermint
Uses the DockerClient to create the network of nodes for testing.
Also, get rid of the http server, as Wiremock can serve the exact same purpose.

patimen committed:
Not quite finished, but important steps towards genesis management
Almost there. Now we can check if the Genesis state matches a `Spec` (which I created with the help of ChatGPT in about 30 minutes).

If it doesn't, we recreate the chain.
What's left is that creating the chain DOESN'T use the Genesis Spec yet, but that should be fairly easy to do.

DimaOrekhovPS reviewed (changes requested):
I think we should keep using docker compose. Could we make DockerGroup.kt spin up clusters using docker compose?

patimen committed:
Merge branch 'main' into launch_from_testermint

patimen committed:
Just use docker compose via command line
DockerClient has no good way to interact with compose, so we just manually fire up docker compose via ProcessBuilder. Saves a ton of duplication, and greatly simplifies what's going on here.

patimen committed:
Add genesis file matching
Now it works: If we assert we need a specific genesis value, we will use the current cluster if it matches, and restart it with the right genesis settings if it doesn't. Woot

patimen committed:
Forgot the changes to the init-genesis.
Two override files. The "default" one and the one created by testermint.
